go.mod

```diff
@@ -26,18 +26,20 @@ require (
 	golang.org/x/crypto v0.0.0-20220315160706-3147a52a75dd
 	golang.org/x/sync v0.1.0
 	golang.org/x/sys v0.4.0
-	golang.org/x/time v0.2.0
+	golang.org/x/time v0.3.0
 	google.golang.org/grpc v1.52.0
-	k8s.io/api v0.23.0
-	k8s.io/apimachinery v0.23.0
-	k8s.io/client-go v0.23.0
+	k8s.io/api v0.25.5
```

Comment:
I think we can update all (include cri-api) to v0.25.6, but I'm not sure if we can update to v0.26.1. Since k8s 1.25 -> 1.26 has a breaking change on cri.

client-go, api, cri-api are all released by k8s release. k8s 1.26+ (for these pkgs, 0.26.0+) drops cri api v1alpha2 support and only support cri api v1.

For cri-api, we need to decide if we want to support v1 only or both v1 and v1alpha2 before updating to 0.26.0+.

For others (client-go, api, apimachinery) (1). we can update to 0.26.0 but that means they're from k8s 1.26 while cri from k8s 1.25. (2) or we can keep all consistent.

Reply:
Does this depend on whether we plan to support k8s (or have we decided not to support k8s in soci)? If it won't be used anymore we can remove those code.

If we haven't decided maybe we can just upgrade to 1.26 and remove v1alpha2 usage. 

Supporting bother versions requires nontrivial changes anyway (from stargz, I feel no difference than removing v1alpha2 and then adding it back... https://github.com/containerd/stargz-snapshotter/pull/1011)

Reply:
Ideally we could share this with stargz. I feel like auth is a building block that most remote snapshotters would need so it feels like a waste to reimplement it or try to keep a fork in sync. 

We should come up with a proposal for how sharing might work and submit it to the stargz folks. Auth seems like a good starting point.

Reply:
In the meantime, we don't have any asks for v1 support, so I'm fine with just going to v0.25.6 for now.